### PR TITLE
fix: harden lifecycle and concurrency paths across core modules

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

Closes several latent lifecycle and concurrency gaps across the kit's core modules. None fire in normal use, but each is a crash or hang risk under DLL unload, hot-reload, or callback misuse.

## Changes

- **logger:** loader-lock-safe `disable_async_mode` leak path with OOM fallbacks, so a detached writer's `AsyncLogger` is never dropped.
- **hook_manager:** mutators and teardown reject reentrant calls from `with_*` callbacks (fail closed) instead of deadlocking; batch toggles log outside the locks.
- **bootstrap:** the detach gate re-arms on a successful attach, making attach/detach cycles re-entrant.
- **worker:** liveness reads from atomics and stop is signaled via a copied `std::stop_source`, off the `jthread` handle.
- **input_intercept:** `install_wndproc` reconciles the predecessor against the swap return to keep a foreign subclasser in the chain.

## Summary by CodeRabbit

* **Documentation**
  * Enhanced thread-safety documentation to clarify reentrancy constraints and shutdown ordering.

* **Bug Fixes**
  * Prevents deadlocks from reentrant calls by gracefully rejecting them instead.
  * Fixed attach/detach cycles to properly reset state.
  * Improved async logger behavior under edge cases.

### CODE REVIEW

File: tests/test_worker.cpp [hunk 1]
```diff
@@ -161,3 +162,50 @@ TEST(StoppableWorker, NameAccessorReturnsConstructionName)
...
+            {
+                while (!poll_stop.load(std::memory_order_acquire))
+                {
+                    (void)w.is_running();
+                    w.request_stop();
+                }
+            });
+    }
+
+    w.shutdown();
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply tests/test_worker.cpp:211]: @coderabbitai

Verified each finding against the current branch; summary below.

### Fixed

- **`hook_manager.hpp` (callback warnings):** Updated the `@warning` blocks on `with_inline_hook`, `with_mid_hook`, and `with_vmt_method` to also forbid a nested `with_*` or `try_with_*` accessor inside a callback, noting each checks the reentrancy guard and fails closed. Doc-only.
- **`test_worker.cpp` (poller start latch):** Added a `std::latch` the pollers count down before looping; the main thread waits on it before `shutdown()`, so the concurrent observation window is actually exercised rather than possibly starting after shutdown finishes.
- **`bootstrap.cpp` (auto-reload on hot-unload):** Confirmed and fixed. `on_logic_dll_unload` and `on_logic_dll_unload_all` now call `Config::disable_auto_reload()` before `clear_registered_items()`, so a watcher whose `on_reload`/setters live in the unloading Logic DLL cannot outlive the unload or leave the next attach reporting `AlreadyRunning`. Added `BootstrapOnLogicDllUnload.StopsAutoReloadWatcher` and `BootstrapOnLogicDllUnloadAll.StopsAutoReloadWatcher`, which assert a fresh `Started` after re-loading.

### Skipped

- **`test_bootstrap.cpp` (join lifecycle thread):** The fixture's assertions (handle cleared after detach, callback counts) depend only on `on_dll_detach`'s synchronous work, not on `DMK_Shutdown` completion. The one shared singleton in play (`Logger`) serializes `shutdown` vs `configure` under its own mutex, so the interleave is benign and independent of the asserted state. There is also no public API to join the internal `lifecycle_thread` (by design `on_dll_detach` must not block under loader lock). The cycle test runs deterministically green across the full suite on both toolchains.
- **`test_hook_manager.cpp` (extract helper / other accessors / live VMT):** The reentrancy guard is a single thread-local counter bumped identically by every `with_*`/`try_with_*` accessor and checked identically by every mutator and teardown path. The existing `Reentrancy_MutatorsFromCallbackFailClosed` already drives the full set from a callback: inline/mid/vmt creators, the `_aob` variants, `hook_vmt_method`, removers, batch toggles, `remove_all_hooks`, and `shutdown`. Re-running through the other accessors re-exercises the same counter, and a non-empty VMT registry adds no coverage because the teardown guards short-circuit before touching the registry.
- **`test_logger.cpp` (leak assertion around `shutdown()`):** `shutdown_internal` and `disable_async_mode` call the identical `leak_async_logger_handle` helper behind the identical `is_loader_lock_held()` gate. `DisableAsyncMode_NoLeakInNormalContext` already pins "normal context takes no leak path" for that shared code, so a parallel assertion on `shutdown()` would re-test the same gate.
